apps/test/netsim/NetSimTable.js

```diff
@@ -0,0 +1,231 @@
+    netsimTable = new NetSimTable(apiTable);
+    callback = function () {};
+    notified = false;
+    netsimTable.tableChangeEvent.register(function () {
+      notified = true;
+    });
+  });
+
+  it ("calls readAll on the API table", function () {
+    netsimTable.readAll(callback);
+    assertEqual(apiTable.log(), 'readAll');
+  });
+
+  it ("calls read on the API table", function () {
+    netsimTable.read(1, callback);
+    assertEqual(apiTable.log(), 'read[1]');
+  });
+
+  it ("calls create on the API table", function () {
+    netsimTable.create({}, callback);
+    assertEqual(apiTable.log(), 'create[{}]');
+  });
+
+  it ("calls update on the API table", function () {
+    netsimTable.update(1, {}, callback);
+    assertEqual(apiTable.log(), 'update[1, {}]');
+  });
+
+  it ("calls delete on the API table", function () {
+    netsimTable.delete(1, callback);
+    assertEqual(apiTable.log(), 'delete[1]');
+  });
+
+  it ("notifies on readAll if any remote row changed", function () {
+    netsimTable.create({data: "A"}, callback);
+
+    notified = false;
+    netsimTable.readAll(callback);
+    assertEqual(notified, false);
+
+    // Remote update - doesn't hit our caches
+    apiTable.update(1, {data: "B"}, callback);
+
+    notified = false;
+    netsimTable.readAll(callback);
+    assertEqual(notified, true);
+  });
+
+  it ("notifies on read if the requested remote row changed", function () {
+    netsimTable.create({data: "A"}, callback);
+
+    notified = false;
+    netsimTable.read(1, callback);
+    assertEqual(notified, false);
+
+    // Remote update - doesn't hit our caches
+    apiTable.update(1, {data: "B"}, callback);
+
+    notified = false;
+    netsimTable.read(1, callback);
+    assertEqual(notified, true);
+  });
+
+  it ("notifies on every create", function () {
+    notified = false;
+    netsimTable.create({}, callback);
+    assertEqual(notified, true);
+  });
+
+  it ("notifies on update if the cache row changed", function () {
+    netsimTable.create({data: "A"}, callback);
+
+    notified = false;
+    netsimTable.update(1, {data: "A"}, callback);
+    assertEqual(notified, false);
+
+    notified = false;
+    netsimTable.update(1, {data: "B"}, callback);
+    assertEqual(notified, true);
+  });
+
+  it ("notifies on delete when row was previously in cache", function () {
+    notified = false;
+    netsimTable.delete(1, callback);
+    assertEqual(notified, false);
+
+    netsimTable.create({}, callback);
+
+    notified = false;
+    netsimTable.delete(1, callback);
+    assertEqual(notified, true);
+  });
+
+  it ("passes new full table contents to notification callbacks", function () {
+    var receivedTableData;
+    netsimTable.tableChangeEvent.register(function (newTableData) {
+      receivedTableData = newTableData;
+    });
+
+    netsimTable.create({data: "A"}, callback);
+    assertEqual(
+        JSON.stringify(receivedTableData),
+        '[{"data":"A","id":1}]');
```

Comment:
I think `_.isEqual` may be a little better here since hashes don't have guaranteed ordering (although they probably will in practice)